managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedCursorImpl.java

```diff
@@ -87,9 +90,10 @@
 
     protected volatile PositionImpl markDeletePosition;
     protected volatile PositionImpl readPosition;
+    private volatile PendingMarkDeleteEntry lastMarkDeleteEntry;
```

Comment:
> is it necessary to introduce lastMarkDeleteEntry field in this PR?
>> The lastMarkDeleteEntry is important to guarantee that we are storing the
same couple of (mark-delete-pos, properties) together, even when we are
rolling over the ledger.

Yes, I removed comment to understand more. But using class `PendingMarkDeleteEntry` for markDeleteEntry is little confusing and hard to set the context. Should we create dedicated sub-class to store this tuple?

Reply:
The `PendingMarkDeleteEntry` is only a container for position plus other fields. I guess we could rename it into `MarkDeleteEntry` or similar.

Reply:
> I guess we could rename it into MarkDeleteEntry or similar.

Yes, renaming would be also fine.